SmartCredentials_aOS/core/src/main/java/de/telekom/smartcredentials/core/api/AuthenticationApi.java

```diff
@@ -33,21 +34,15 @@
 
     /**
      * Initializes all necessary dependencies for specified provider.
-     * This will fetch an OpenID Connect discovery document
-     * from the issuer in the configuration to configure this instance for use
-     * or will use the endpoints specified in the configuration document.
+     * This will fetch an OpenID Connect discovery document from the issuer in the configuration
+     * to configure this instance for use or will use the endpoints specified in the configuration
+     * document.
      *
-     * @param context           The application context
-     * @param customTabBarColor The color of the custom tab's action bar
-     * @param listener          An {@link AuthenticationServiceInitListener} that will be called
-     *                          once the initialization is complete or fail
+     * @param authenticationConfiguration The application configuration file which contains the
+     *                                    parameters used for the authentication flow.
      */
     @SuppressWarnings("unused")
-    SmartCredentialsApiResponse<Boolean> initialize(Context context,
-                    String identityProviderId,
-                    int authConfigFileResId,
-                    int customTabBarColor,
-                    AuthenticationServiceInitListener listener);
+    SmartCredentialsApiResponse<Boolean> initialize(AuthenticationConfiguration authenticationConfiguration);
```

Comment:
the listener should be kept outside the configuration, because is not a configuration. Debatable for the context ... but I would incline to extracting it out of the configuration.

Reply:
Removed the listener from the configuration